lib/rex/post/meterpreter/packet_dispatcher.rb

```diff
@@ -130,8 +131,19 @@ def send_packet(packet, opts={})
       tlv_enc_key = opts[:tlv_enc_key]
     end
 
-    # Uncomment this line if you want to see outbound packets in the console.
-    # STDERR.puts("\n\e[1;31mSEND\e[0m: #{packet.inspect}\n")
+    if framework.datastore['sessiontlvlogging']
+      if framework.datastore['sessiontlvlogging'].downcase == 'console'
+        print "\n%redSEND%clr: #{packet.inspect}\n"
+      elsif framework.datastore['sessiontlvlogging'].downcase =~ /^file:/
+        path = framework.datastore['sessiontlvlogging'].downcase.split('file:').last
+        pathname = ::Pathname.new(path)
+        if pathname.directory?
+          ::File.open(::File.dirname(pathname) + '/sessiontlvlogging.txt', 'a+') { |f| f.puts("\nSEND: #{packet.inspect}\n") }
+        elsif pathname.file? || !pathname.exist?
+          ::File.open(pathname, 'a+') { |f| f.puts("\nSEND: #{packet.inspect}\n") }
```

Comment:
Opening and closing files as part of a logging loop will have performance hit - the common pattern is to usually open a file once - then write to the open file when required 👀